typing/ctype.ml

```diff
@@ -7356,3 +7356,82 @@ let constrain_decl_jkind env decl jkind =
         match decl.type_manifest with
         | None -> err
         | Some ty -> constrain_type_jkind env ty jkind
+
+type 'res constructor_crossing_kind =
+  | Creation : Mode.Value.r constructor_crossing_kind
+  | Destruction : Mode.Value.l constructor_crossing_kind
+  | Rebinding : unit constructor_crossing_kind
+
+let check_exn_constructor_crossing (type res)
```

Comment:
looking at this function, the code sharing between `Creation` and `Destruction` is quite limited - maybe better to just have two functions. And `Rebinding` just calls both.

Reply:
Done.